### PR TITLE
New Data Source `azurerm_batch_application`

### PR DESCRIPTION
**Pull request by oWretch**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -40,6 +40,7 @@ FEATURES:
 
 * **New Data Source:** `azurerm_eventgrid_system_topic` ([#13851](https://github.com/hashicorp/terraform-provider-azurerm/issues/13851))
 * **New Data Source:** `azurerm_billing_mpa_account_scope` ([#13723](https://github.com/hashicorp/terraform-provider-azurerm/issues/13723))
+* **New Data Source:** `azurerm_batch_application` ([#14043](https://github.com/hashicorp/terraform-provider-azurerm/issues/14043))
```
Comment (stephybun) [inline CHANGELOG.md:43]: While the foresight is appreciated here, the changelog entry is only done after the PR has been approved and merged.
```suggestion
```

File: internal/services/batch/batch_application_data_source.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+}
+
+func dataSourceBatchApplicationRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Batch.ApplicationClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	resourceGroup := d.Get("resource_group_name").(string)
+	name := d.Get("name").(string)
+	accountName := d.Get("account_name").(string)
```
Comment (stephybun) [inline internal/services/batch/batch_application_data_source.go:63]: Could we use the id parser here
```suggestion
        subscriptionId := meta.(*clients.Client).Account.SubscriptionId
        id := parse.NewApplicationID(subscriptionId, d.Get("resource_group_name").(string), d.Get("account_name").(string), d.Get("name").(string))
```

File: internal/services/batch/batch_application_data_source.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+	resp, err := client.Get(ctx, resourceGroup, accountName, name)
```
Comment (stephybun) [inline internal/services/batch/batch_application_data_source.go:65]: Any references to resource group, account name or name can be replaced with the values from the id
```suggestion
	resp, err := client.Get(ctx, id.ResourceGroup id.BatchAccountName, id.Name)
```

File: internal/services/batch/batch_application_data_source.go [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			return fmt.Errorf("Error: Batch Application %q (Resource Group %q) was not found", name, resourceGroup)
```
Comment (stephybun) [inline internal/services/batch/batch_application_data_source.go:68]: When the `id` is printed it contains all of this information, so the error messages can become more succinct
```suggestion
			return fmt.Errorf("%s was not found", id)
```

File: internal/services/batch/batch_application_data_source.go [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+		}
+		return fmt.Errorf("making Read request on AzureRM Batch Application %q (Account Name %q / dataSource Group %q): %+v", name, accountName, resourceGroup, err)
```
Comment (stephybun) [inline internal/services/batch/batch_application_data_source.go:70]:
```suggestion
		return fmt.Errorf("reading %s: %+v", id, err)
```

File: internal/services/batch/batch_application_data_source.go [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+	}
+
+	d.SetId(*resp.ID)
```
Comment (stephybun) [inline internal/services/batch/batch_application_data_source.go:73]: We'll set the id here with the one from the parser.
```suggestion
	d.SetId(id.ID())
```

File: website/docs/d/batch_application.html.markdown [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+output "batch_application_id" {
+  value = data.azurerm_batch_application.example.id
+}
+```
+
+## Argument Reference
+
+* `name` - The name of the Application.
+
+* `resource_group_name` - The Name of the Resource Group where this Batch account exists.
```
Comment (stephybun) [inline website/docs/d/batch_application.html.markdown:31]:
```suggestion
* `resource_group_name` - The name of the Resource Group where this Batch account exists.
```

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks for this new data source @oWretch. Overall this is in good shape, there are just a few minor suggestions 🙂

**Comment by oWretch** [comment]
Thanks @stephybun for the great suggestions. I see more work ahead to take those and back-port them to the existing definitions in the batch service definitions :)

**Comment by stephybun** [comment]
Thanks @oWretch, LGTM 👍🏼

**Review by stephybun — APPROVED** [review]

**Pull request merged**